src/mesh/Utils.hpp

```diff
@@ -0,0 +1,140 @@
+#pragma once
+
+#include <algorithm>
+#include <array>
+#include <mesh/Edge.hpp>
+#include <mesh/Mesh.hpp>
+#include <utility>
+
+namespace precice {
+namespace mesh {
+
+/** return a pointer to the shared vertex of 2 edges
+ * @param[in] a pointer to Edge a
+ * @param[in] a pointer to Edge a
+ * @returns a pointer to the Vertex shared by a and b or nullptr otherwise.
+ */
+inline Vertex *sharedVertex(Edge &a, Edge &b)
+{
+  Vertex *a0 = &a.vertex(0);
+  Vertex *a1 = &a.vertex(1);
+  Vertex *b0 = &b.vertex(0);
+  Vertex *b1 = &b.vertex(1);
+  if (a0 == b0 || a0 == b1) {
+    return a0;
+  }
+  if (a1 == b0 || a1 == b1) {
+    return a1;
+  }
+  return nullptr;
+}
+
+/** Calulates the length of an Edge
+ *
+ * @param[in] e the edge
+ *
+ * @returns the distance between both vertices of e 
+ */
+inline double edgeLength(const Edge& e)
+{
+  return (e.vertex(0).getCoords() - e.vertex(1).getCoords()).norm();
+}
+
+template <std::size_t n>
+struct Chain {
+  /// true if the chain is connected or closed and thus valid
+  bool connected;
+  /// undefined if not connected
+  std::array<Vertex *, n> vertices;
+  /// undefined if not connected
+  std::array<Edge *, n> edges;
+};
+
+/** Generates a chain for an array of edges.
+ *
+ * The resulting verices and edges are undefined if the chain is not connected.
+ * If the edges form a chain, then the first edge of the resulting chain is the first edge of the argument.
+ * Also, the first and last vertex of the chain will be the vertices of the first edge of the argument.
+ *
+ * @param[in] edges an array of pointers to edges to chain together
+ *
+ * @returns A \ref Chain of the input
+ */
+template <std::size_t n>
+Chain<n> asChain(std::array<mesh::Edge *, n> edges)
+{
+  static_assert(n > 1, "You already know the answer.");
```

Comment:
Why do you use `static_assert` here?

Reply:
It tells a silly developer off at compile time.